pipeline/clean/opuscleaner/clean-corpus.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+##
+# Cleaning parallel corpora with OpusCleaner
+#
+
+set -x
+set -euo pipefail
+
+echo "###### Cleaning corpus with OpusCleaner"
+
+test -v SRC
+test -v TRG
+
+input_prefix=$1
+output_prefix=$2
+threads=$3
+dataset=$4
+
+COMPRESSION_CMD="${COMPRESSION_CMD:-pigz}"
+ARTIFACT_EXT="${ARTIFACT_EXT:-gz}"
+
+if [ "$threads" = "auto" ]; then
+  threads=$(nproc)
+fi
+
+cd "$(dirname "${0}")"
+dir="$(dirname "${output_prefix}")"
+mkdir -p "${dir}"
+
+echo "### Generating cleaning config: ${dataset}.${SRC}-${TRG}.filters.json"
+# save new filter to dataset output dir
+filter_path="${output_prefix}.${SRC}-${TRG}.filters.json"
+python3 generate_filters.py "${input_prefix}" "${SRC}" "${TRG}" "${dataset}" "${filter_path}"
+test -s "${filter_path}" || exit 1
+
+echo "### Cleaning ${input_prefix} with filter ${filter_path}"
+paste <(${COMPRESSION_CMD} -dc "${input_prefix}.${SRC}.${ARTIFACT_EXT}") \
+      <(${COMPRESSION_CMD} -dc "${input_prefix}.${TRG}.${ARTIFACT_EXT}") |
+opuscleaner-clean \
+  --parallel ${threads} \
+  --batch-size=50000 \
+  --input=- \
+  "${filter_path}" "${SRC}" "${TRG}" \
+  2> "${output_prefix}.opus_filter.log" |
```

Comment:
Question: I guess this outputs the opus log to the artifacts directory.

Reply:
Yes, sometimes the opuscleaner writes to stderr that we can't see in the artifacts. We should address this issue on a higher level probably.